### ISSUE TITLE
React Router with plugin not working

### ISSUE THREAD

**Issue by mnlbox** [oedotme/generouted#45]
I just followed [this guide ](https://github.com/oedotme/generouted/tree/main/plugins/react-router) and [this example](https://github.com/oedotme/generouted/tree/main/examples/react-router/plugin) but I have a couple of problems in this setup.

1. When I try to install the plugin I had the error below related to `fast-glob` package. I think installing this package is missing in the guide:
```
pnpm add -D @generouted/react-router
Packages: +1
+
Progress: resolved 166, reused 144, downloaded 0, added 0, done

devDependencies:
+ @generouted/react-router 1.7.5

 WARN  Issues with peer dependencies found
.
└─┬ @generouted/react-router 1.7.5
  └── ✕ missing peer fast-glob@>=3
Peer dependencies that should be installed:
  fast-glob@>=3  

Done in 2s
```

Then I tried to install `fast-glob` manually like this to solve this step and it seems it's working:
```
pnpm add -D fast-glob
```
2. Then I got a type error on my `vite.config.ts` file as below:
![01](https://user-images.githubusercontent.com/25859075/218244431-c1901ec3-3799-409c-803f-41250c535f53.png)
But it seems the `routes.gen.tsx` file is created for me in the `src` folder so I think we can ignore this issue for now.

3. I just added some pages exactly same as the example but I only can see blank pages as a result in the browser.
After switch from plugin setup to the normal setup everything is working without any other changes anywhere. I just replaced this line:
```
import { Routes } from './routes.gen'
```
with this:
```
import { Routes } from 'generouted/react-router'
```
and installed `generouted` package rather than `@generouted/react-router`

**Comment by brotherko** [comment]
I had the exact same issue with you before. Make sure you have to use the <Outlet /> component inside your _app file instead of children

**Comment by oedotme** [comment]
Hey @mnlbox, Thanks for providing all the details!

`1` I wanted to ask which `vite` version are you using?

`2` The types should work fine now in `vite.config.ts`, it was an issue solved in [v1.7.6](https://github.com/oedotme/generouted/releases/tag/v1.7.6)

`3` As @brotherko pointed out, there's an issue that requires having `pages/_app.tsx` file and renders an `<Outlet />` component from React Router:

```tsx
// src/pages/_app.tsx

import { Outlet } from 'react-router-dom'

export default function App() {
  return <Outlet />
}
```

I'll close this issue after solving issue `3`, which is expected as `pages/_app.tsx` is optional.

**Comment by mnlbox** [comment]
Hi @oedotme,

Thanks for your reply. My vite version is `4.1.0`.
Ah, good point about that `<Outlet />` thanks for the clarification.

**Comment by oedotme** [comment]
@mnlbox Not having a `pages/_app.tsx` file should also work fine now [v1.7.7](https://github.com/oedotme/generouted/releases/tag/v1.7.7).

But in case of using `pages/_app.tsx`, you'll still need an `<Outlet />` component to render the children, and I added an example for the optional root layout to the readme.

Lastly for the missing `fast-glob` installation is fixed at [v1.7.8](https://github.com/oedotme/generouted/releases/tag/v1.7.8), it should be now installed automatically while installing `@generouted/react-router`.

Please let me know if all works now as expected!
